java/src/test/java/com/ncc/aif/ExamplesAndValidationTest.java

```diff
@@ -1355,6 +1356,25 @@ void relationOfUnknownType() {
             utils.testInvalid("Invalid: relation of unknown type");
         }
 
+        @Test
+        void clusterPrototypeMissingTypeAssertion() {
```

Comment:
I'm not sure how this is not covered by entityMissingType(). I agree that it's not explicitly covering this case, but logically, I think it does. Can you think of a situation in which entityMissingType() would pass and clusterPrototypeMissingTypeAssertion would fail?

Reply:
I agree that if a cluster prototype is missing a type assertion, then `entityMissingType()` will also fail, even though `entityMissingType()` isn't testing entity clusters.

I think I was being especially conservative/explicit (and arguably redundant) in this test because the impact of the query change is so immense.  I wanted to know beyond a shadow of a doubt that it was perfectly valid, by explicitly covering all (possible) cases in the truth table.

But that said, if you prefer that the test be removed, I will do so.  It will not catch anything that `entityMissingType()` won't catch.

Reply:
Yeah, if that's the case, I would recommend removing the test. If you feel strongly that it conveys something to the researchers, we can leave it in

Reply:
All things being equal, I would have left it in, but I don't feel strongly about it, nor does it convey anything important to the researchers.  The important thing is that we've validated that the new query work in all cases.  Test removed.